File: 4. Largest Palindrome Product/solution.go
package main

import (
	"fmt"
	"strconv"
)

func isPalindrome(n int) bool {
	strN := strconv.Itoa(n)
	reversed := ""
	for i := len(strN) - 1; i >= 0; i-- {
		reversed += string(strN[i])
	}
	return strN == reversed
}

func largestPalindrome() int {
	for num := 999 * 999; num >= 100*100; num-- {
		if isPalindrome(num) {
			for i := 999; i >= 100; i-- {
				if num%i == 0 && 100 <= num/i && num/i <= 999 {
					return num
				}
			}
		}
	}
	return -1
}

func main() {
	fmt.Println(largestPalindrome())
}

File: 4. Largest Palindrome Product/solution.py
def is_palindrome(n):
    return str(n) == str(n)[::-1]

def largest_palindrome():
    for num in range(999*999, 100*100, -1):
        if is_palindrome(num):
            for i in range(999, 99, -1):
                if num % i == 0 and 100 <= num // i <= 999:
                    return num

print(largest_palindrome())